### ISSUE TITLE
In split view, editor and preview views do not scroll in sync

### ISSUE THREAD

**Issue by guillerodriguez** [mb21/panwriter#95]
The preview and editor views do not scroll in sync; if you scroll any of the two views, the other view will also scroll, but not at the same pace or not to the same location, which sort of defeats the purpose of the split view.

This is Panwriter 0.8.2 on macOS / M1.

**Comment by Skeeve** [comment]
Still present in 0.8.5


I'm using

- Mac OS 13.1
- pandoc 3.1
- panWriter 0.8.5

**Comment by micha06de** [comment]
For me, version 0.8.6 on Windows it is even worse, the windows are not only not synchronised, if I am in the preview window somewhere in the document and scroll in the Markdown window (no matter in which direction) the document in the preview window jumps to the top of the document.

Is there a workaround (I don't think it was always like this).

**Comment by mb21** [comment]
Yeah, on some documents I can reproduce it. I'm pretty sure it used to work a lot better, probably broke in some Electron (i.e. Chromium) upgrade. Anyway, the [code is somewhat finicky](https://github.com/mb21/panwriter/blob/master/src/renderPreview/scrolling.ts) and to be frank, I've been putting this off thinking we should [upgrade to CodeMirror 6](https://codemirror.net/docs/migration/) first. If anybody would like to start working on that, pull requests welcome!

**Comment by riuttner** [comment]
Can confirm too. This is especially a problem as - at least on Windows - a dynamic overlay window pops up if you are at the top of the visible area. Thus you cannot even reach the scrollbar buttons, if your file is large, because they are fully covered. Instead you have to somehow act for a while on your scroll wheel until you hopefully find a way to really scroll. So please either fix the scrolling in general soon or at least add an option to disable the automatic overlay window (e.g. let it disappear after 1 second), because in big files scrolling is almost impossible to be started now.

**Comment by mb21** [comment]
I'd be happy to review well-tested pull request. Either by finishing https://github.com/mb21/panwriter/pull/170 first, or just doing https://github.com/mb21/panwriter/pull/192 for now. Unfortunately, personally I don't have the time right now to implement and thoroughly test this, but happy to answer any questions.

**Comment by riuttner** [comment]
I understand that it is quite some effort. As I personally have no experience with typescript and do not know others having that, I would be interested to find a developer who already did some work around panwriter and can do that (during the next 2 months or so). Especially as this is something also requested by other users, I would like to fund that work. For contacting me, just write to the contact mail at my company address.

**Comment by mb21** [comment]
@riuttner Thanks! Are you interested in sponsoring the upgrade form CodeMirror 5 to 6 (which would be a bigger thing but long-term for the best), or just the scroll sync issue with current CodeMirror 5?

**Comment by riuttner** [comment]
@mb21 You are welcome ... but of course it depends on the concrete amount needed to do that. As I personally see a long-term perspective for my interest in Panwriter, I would prefer the upgrade from 5 to 6. Can you estimate the effort difference as a rough factor?
